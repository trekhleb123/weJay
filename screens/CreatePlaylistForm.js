import React, { useState, useEffect } from 'react';
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableOpacity
} from 'react-native';
import { TextInput } from 'react-native-gesture-handler';
import { getTokens, logIn, makeNewPlaylist } from '../api/spotify';
import { createRoom } from '../firebase/index';

export default function CreatePlaylistForm(props) {
  const [roomData, setRoomData] = useState({
    title: '',
    hostName: '',
    passcode: null
  });
  const handleSubmit = async formData => {
    try {
      const spotifyTokens = await getTokens();
      formData.accessToken = spotifyTokens.access_token;
      formData.refreshToken = spotifyTokens.refresh_token;
      formData.expiresIn = spotifyTokens.expires_in;
      //create room in firebase with Data, add playlist id
      //make playlist on spotify
      formData.playlistID = await makeNewPlaylist(
        formData.accessToken,
        formData.title
      );
      let docId = await createRoom(formData);
      if (formData.accessToken) {
        props.navigation.navigate('PlaylistRoom', { docId: docId });
      }
    } catch (err) {
      console.log(err);
    }
  };
  return (
    <View style={styles.container}>
      <View>
        <Text style={styles.header}>Create Your Playlist</Text>
      </View>
        <View style={styles.inputContainer}>
          <TextInput
            style={styles.textInput}
            placeholder="Playlist Title"
            maxLength={100}
            value={roomData.title}
            onChangeText={text => setRoomData({ ...roomData, title: text })}
          />
          <TextInput
            style={styles.textInput}
            placeholder="Your Name"
            maxLength={50}
            value={roomData.hostName}
            onChangeText={text => setRoomData({ ...roomData, hostName: text })}
          />
          <TextInput
            style={styles.textInput}
            placeholder="Your Playlist Passcode"
            maxLength={4}
            value={roomData.passcode}
            onChangeText={text => setRoomData({ ...roomData, passcode: text })}
          />
        </View>
        <TouchableOpacity
          style={styles.saveButton}
          onPress={() => handleSubmit(roomData)}
        >
          <Text style={styles.saveButtonText}>Create Playlist</Text>
        </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#F5FCFF'
  },
  header: {
    fontSize: 25,
    textAlign: 'center',
    fontWeight: 'bold'
  },
  inputContainer: {
    paddingTop: 15,
    marginLeft: 40,
  },
  textInput: {
    borderColor: '#fff',
    backgroundColor: '#fff',
    borderRadius:30,
    borderWidth: 1,
    width:250,
    fontSize: 18,
    height:45,
    textAlign:'center',
    marginBottom:20,
    flexDirection: 'row',
    alignItems:'center',
    shadowColor: '#999',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.5,
    shadowRadius: 2,
    elevation: 1,
  },
  saveButton: {
    backgroundColor: '#FF5857',
    padding: 15,
    borderRadius: 25,
    width:200,
    marginLeft: 65,
    shadowColor: '#999',
    shadowOffset: { width: 1, height: 2 },
    shadowOpacity: 0.5,
    shadowRadius: 2,
  },
  saveButtonText: {
    color: '#FFFFFF',
    fontSize: 20,
    textAlign: 'center'
  }
});
