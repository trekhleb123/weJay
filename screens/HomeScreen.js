import React, { useState } from 'react';
import {
  Text,
  StyleSheet,
  TouchableHighlight,
  TouchableOpacity,
  View,
  StatusBar
} from 'react-native';

export default function HomeScreen(props) {
  const [docId, setDocId] = useState('');
  const [userName, setUserName] = useState('');
  return (
    <View style={styles.container}>
      <StatusBar barStyle="dark-content" backgroundColor="#FF5857" />
      <View style={styles.header}>
        <Text style={{ fontSize: 40, fontWeight: 'bold', alignSelf: 'center' }}>
          weJay
        </Text>
      </View>
      <View>
        <TouchableHighlight
          style={styles.button}
          onPress={() =>
            props.navigation.navigate('CreatePlaylistForm', {
              otherParam: 'Create a Playlist',
              docId, setDocId, userName,setUserName
            })
          }
        >
          <Text style={styles.text}>Create Playlist</Text>
        </TouchableHighlight>

        <TouchableOpacity
          style={styles.button}
          onPress={() =>
            props.navigation.navigate('JoinPlaylistForm', {
              docId,
              setDocId,
              userName,
              setUserName,
              otherParam: 'Join A Playlist'
            })
          }
        >
          <Text style={styles.text}>Join Playlist</Text>
        </TouchableOpacity>
          { docId ?( <TouchableOpacity
            style={styles.button}
            onPress={() => props.navigation.navigate('PlaylistRoom', {docId, setDocId, userName,setUserName})}
          >
            <Text style={styles.text}>Go To Playlist</Text>
          </TouchableOpacity>)
         : null}

        {/* NEED THIS TO TEST, WILL REMOVE AFTER TESTING */}

        <TouchableOpacity
          style={styles.button}
          onPress={() => props.navigation.navigate('Playback')}
        >
          <Text style={styles.text}>Playback (TEST)</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: '#F4F8FF'
  },
  header: {
    paddingBottom: 25,
    marginTop: 80
  },
  text: {
    fontSize: 24,
    textAlign: 'center',
    fontWeight: 'bold',
    color: '#fff'
  },
  button: {
    marginTop: 10,
    shadowColor: '#999',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.5,
    shadowRadius: 2,
    elevation: 1,
    borderRadius: 25,
    padding: 10,
    backgroundColor: '#FF5857',
    width: 300
  }
});
