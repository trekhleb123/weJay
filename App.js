// THIS IS TEST CODE IP, UNALTERED BOILER PLATE CODE CAN BE FOUND IN APPTEST.JS

'use strict';
import React, { Component } from 'react';
import { AppRegistry } from 'react-native';
import { createStackNavigator, createAppContainer } from 'react-navigation';

import HomeScreen from './screens/HomeScreen';
import CreatePlaylistForm from './screens/CreatePlaylistForm';
import JoinPlaylistForm from './screens/JoinPlaylistForm';
import PlaylistRoom from './screens/PlaylistRoom';
import SearchScreen from './screens/SearchScreen';
import CleanSearch from './screens/CleanSearch';
import TestRoom from './screens/TestRoom';
//if you want to use our keys you need to require the secrets file
require('./secrets')
export default class ReactNav extends Component {
  render() {
    return <App />;
  }
}

// class GreenScreen extends Component {
//   static navigationOptions = {
//     title: 'Green'
//   };
//   render() {
//     return (
//       <View style={styles.green}>
//         <Text style={styles.text}>This is the Green Screen</Text>
//         <TouchableHighlight
//           style={styles.button}
//           onPress={() => this.props.navigation.navigate('Red')}
//         >
//           <Text style={styles.text}>Go to Red</Text>
//         </TouchableHighlight>
//       </View>
//     );
//   }
// }

// class RedScreen extends Component {
//   render() {
//     return (
//       <View style={styles.red}>
//         <Text style={styles.text}>This is the Red Screen</Text>
//         <TouchableHighlight
//           style={styles.button}
//           onPress={() => this.props.navigation.goBack()}
//         >
//           <Text style={styles.text}>Back to Green</Text>
//         </TouchableHighlight>
//       </View>
//     );
//   }
// }

// RedScreen.navigationOptions = props => {
//   const { navigation } = props;
//   return {
//     headerTitle: 'Red',
//     headerRight: (
//       <Button title="Purple" onPress={() => navigation.navigate('Purple')} />
//     ),
//     headerLeft: (
//       <Button title="Blue" onPress={() => navigation.navigate('Blue')} />
//     )
//   };
// };

// class BlueScreen extends Component {
//   render() {
//     return (
//       <View style={styles.blue}>
//         <Text style={styles.text}>This is the Blue Screen</Text>
//       </View>
//     );
//   }
// }

// class PurpleScreen extends Component {
//   render() {
//     return (
//       <View style={styles.purple}>
//         <Text style={styles.text}>This is the Purple Screen</Text>
//       </View>
//     );
//   }
// }

const StackNav = createStackNavigator({
  Home: { screen: HomeScreen },
  CreatePlaylistForm: { screen: CreatePlaylistForm },
  JoinPlaylistForm: { screen: JoinPlaylistForm },
  PlaylistRoom: { screen: PlaylistRoom },
  SearchScreen: { screen: SearchScreen },
  TestRoom: { screen: TestRoom },
  CleanSearch: { screen: CleanSearch}
});

const App = createAppContainer(StackNav);

AppRegistry.registerComponent('ReactNav', () => ReactNav);
