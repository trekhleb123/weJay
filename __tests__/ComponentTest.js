import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import CreatePlaylistForm from '../screens/CreatePlaylistForm'
import HomeScreen from '../screens/HomeScreen';
import { Provider } from "react-redux";
import configureMockStore from 'redux-mock-store'

const mockStore = configureMockStore()

describe('Component Snapshots', () => {

  it(`App.js renders correctly`, () => {
    const tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('CreatePlaylistForm renders correctly', () => {
    const store = mockStore({});
    const tree =renderer.create(
        <Provider store ={store}>
            <CreatePlaylistForm/>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`HomeScreen renders correctly`, () => {
    const store = mockStore({})
    const tree = renderer.create(
      <Provider store={store}>
          <HomeScreen/>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
