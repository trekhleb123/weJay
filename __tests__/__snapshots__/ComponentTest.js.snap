// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the loading screen 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "column-reverse",
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "height": 1334,
                "justifyContent": "flex-start",
                "width": 750,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "height": 1334,
                  "width": 750,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://s30226.pcdn.co/wp-content/uploads/2015/02/half-moon-party-dates.jpg",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 1334,
                      "width": 750,
                    },
                    Object {
                      "opacity": 0.7,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <Image
                  resizeMode="contain"
                  source={1}
                  style={
                    Object {
                      "height": 533.5,
                      "marginTop": 200,
                      "width": 525,
                    }
                  }
                />
                <View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#FF5857",
                        "borderRadius": 25,
                        "marginTop": 30,
                        "padding": 10,
                        "width": 300,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 24,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        }
                      }
                    >
                      Create Playlist
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#FF5857",
                        "borderRadius": 25,
                        "marginTop": 30,
                        "opacity": 1,
                        "padding": 10,
                        "width": 300,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 24,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        }
                      }
                    >
                      Join Playlist
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "transform": Array [
              Object {
                "translateX": -750,
              },
            ],
          }
        }
      >
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "#fff",
              "borderBottomColor": "#a7a7aa",
              "borderBottomWidth": 0.5,
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`App renders the root without loading screen 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "column-reverse",
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "height": 1334,
                "justifyContent": "flex-start",
                "width": 750,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "height": 1334,
                  "width": 750,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://s30226.pcdn.co/wp-content/uploads/2015/02/half-moon-party-dates.jpg",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 1334,
                      "width": 750,
                    },
                    Object {
                      "opacity": 0.7,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <Image
                  resizeMode="contain"
                  source={1}
                  style={
                    Object {
                      "height": 533.5,
                      "marginTop": 200,
                      "width": 525,
                    }
                  }
                />
                <View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#FF5857",
                        "borderRadius": 25,
                        "marginTop": 30,
                        "padding": 10,
                        "width": 300,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 24,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        }
                      }
                    >
                      Create Playlist
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#FF5857",
                        "borderRadius": 25,
                        "marginTop": 30,
                        "opacity": 1,
                        "padding": 10,
                        "width": 300,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fff",
                          "fontSize": 24,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        }
                      }
                    >
                      Join Playlist
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "transform": Array [
              Object {
                "translateX": -750,
              },
            ],
          }
        }
      >
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "#fff",
              "borderBottomColor": "#a7a7aa",
              "borderBottomWidth": 0.5,
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
